### ISSUE TITLE
[0.28.0] type error in editor.api.d.ts

### ISSUE THREAD

**Issue by AviVahl** [microsoft/monaco-editor#2669]
**monaco-editor version:** 0.28.0
**Browser:** Irrelevant
**OS:** Fedora 34

When type checking against the new version, typescript gave the following type error:
```
node_modules/monaco-editor/esm/vs/editor/editor.api.d.ts:16:25 - error TS2503: Cannot find namespace 'monaco'.

16     MonacoEnvironment?: monaco.Environment | undefined;
                           ~~~~~~


Found 1 error.
```
Should be noted that our project has `"skipLibCheck"` commented out in `tsconfig.json`, so we are type checking involved `.d.ts` files as well.

**Comment by AviVahl** [comment]
Exact steps to reproduce:
```
mkdir monaco-type-error
cd monaco-type-error
npm init -y
npm i typescript monaco-editor
npx tsc --init
<edit tsconfig.json with your favorite editor; comment out skipLibCheck on that last line>
echo "import * as monaco from 'monaco-editor'; console.log(monaco)" > a.ts
npx tsc
```

**Comment by troy351** [comment]
This is a critical issue @hediet

**Comment by hediet** [comment]
Should be fixed with 0.28.1. Could you verify that?

**Comment by AviVahl** [comment]
Verified on my end. Thank you! :+1:
